samples/nrf9160/azure_iot_hub/src/main.c

```diff
@@ -124,17 +129,21 @@ static void on_evt_twin_desired(char *buf, size_t len)
 
 static void on_evt_direct_method(struct azure_iot_hub_method *method)
 {
-	printk("Method name: %s\n", method->name);
-	printf("Payload: %.*s\n", method->payload_len, method->payload);
+	size_t request_id_len = MIN(sizeof(method_data.request_id) - 1, method->request_id.size);
+	size_t name_len = MIN(sizeof(method_data.name) - 1, method->name.size);
+
+	printk("Method name: %.*s\n", method->name.size, method->name.ptr);
+	printk("Payload: %.*s\n", method->payload.size, method->payload.ptr);
+
+	memcpy(method_data.request_id, method->request_id.ptr, request_id_len);
 
-	strncpy(method_data.request_id, method->rid, sizeof(method_data.request_id) - 1);
-	method_data.request_id[sizeof(method_data.request_id) - 1] = '\0';
+	method_data.request_id[request_id_len] = '\0';
```

Comment:
If we add `request_id_len` to `method_data`, we can avoid declaring the local variable and the string termination here. Then just change the logic in `direct_method_handler` to use the already calculated value instead of calculating lenght based on the string terminator.

And perhaps do the same for name even though the length isn't used in `direct_method_handler`?